storm/src/main/storm/mesos/resources/ResourceEntries.java

```diff
@@ -93,15 +93,15 @@ public ReservationType getReservationType() {
     public RangeResourceEntry add(ResourceEntry<Long> resourceEntry) {
```

Comment:
@dsKarthick : can you please explain *exactly* what your logic / vision is for all of this RangeResource stuff?  My understanding from briefly reading this code is that as we are adding an Offer's port resources (which could have say ports: `[1, 21-23, 10]` ) to the existing aggregated port resources from this host (which could be say `[15-17, 25, 40-60]` ).     But I'm not clear how you're then merging the new offer's list of port resources into the existing aggregated ports.